testutils/inject/shell_service.go

```diff
@@ -22,3 +26,32 @@ func (s *ShellService) DoCommand(ctx context.Context,
 	}
 	return s.DoCommandFunc(ctx, cmd)
 }
+
+// ShellServiceWithReconfigure represents a fake instance of a shell service.
```

Comment:
Eric brought this up in slack the other day. We want to avoid mocking resources directly in test files, and instead use (and extend) this testutils/inject library..